### PR TITLE
AP/MIN-462: add total borrow rpc

### PR DESCRIPTION
**Pull request by aparkalov**
**Description:**
Add total borrow in usd per user across all pools RPC,
Add RPC tests 

**Changes in storage or API:**

API changes: getUserTotalBorrowToUsd RPC added, see [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [x] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

File: runtime/src/rpc_tests.rs [hunk 1]
```diff
@@ -1875,3 +1881,64 @@ fn get_user_data_rpc_should_work() {
...
+		.execute_with(|| {
+			assert_ok!(set_oracle_price_for_all_pools(2));
+			assert_ok!(MinterestProtocol::deposit_underlying(alice(), DOT, 100_000 * DOLLARS));
+			assert_ok!(MinterestProtocol::deposit_underlying(alice(), ETH, 100_000 * DOLLARS));
+			assert_ok!(MinterestProtocol::enable_is_collateral(alice(), DOT));
+			assert_ok!(MinterestProtocol::enable_is_collateral(alice(), ETH));
+			assert_ok!(MinterestProtocol::borrow(alice(), ETH, 80_000 * DOLLARS));
+			assert_ok!(MinterestProtocol::borrow(alice(), DOT, 50_000 * DOLLARS));
+
+			// in the block when borrow happens there should be no accrued dust
```
Comment (Apollo50) [inline runtime/src/rpc_tests.rs:1902]:
```suggestion
			// in the block when borrow happens there should be no interest accumulated
```

File: runtime/src/rpc_tests.rs [hunk 2]
```diff
@@ -1875,3 +1881,64 @@ fn get_user_data_rpc_should_work() {
...
+			// TODO: add formulas
+			assert_eq!(
+				get_user_total_borrow_usd_rpc(ALICE::get()),
+				130000016020000000000000u128
+			);
+		})
+}
+
+#[test]
+fn get_user_total_borrow_usd_rpc_without_borrowed_assets_should_work() {
```
[the remark this answers is not in the document]
Comment (aparkalov) [reply runtime/src/rpc_tests.rs]: fixed

File: pallets/controller/rpc/src/lib.rs [hunk 3]
```diff
@@ -169,6 +169,17 @@ pub trait ControllerRpcApi<BlockHash, AccountId> {
...
 	fn get_user_data(&self, account_id: AccountId, at: Option<BlockHash>) -> Result<Option<UserData>>;
+
+	/// Returns total borrow balance for user per all assets based on fresh latest indexes.
+	///
+	///  - `&self` :  Self reference
+	///  - `account_id`: current account id.
+	///  - `at` : Needed for runtime API use. Runtime API must always be called at a specific block.
+	///
+	/// Return:
+	/// - amount: account total borrow in usd.
```
Comment (lut1k) [inline pallets/controller/rpc/src/lib.rs:180]: In MIN-447 comments for RPC are formatted differently:
https://github.com/minterest-finance/minterest-chain-node/pull/168/files

File: pallets/controller/src/lib.rs [hunk 4]
```diff
@@ -596,6 +596,25 @@ impl<T: Config> Pallet<T> {
 			None => true,
 		}
 	}
+
+	/// Calculate user total borrow in usd based on fresh exchange rate and
+	/// latest oracle price
+	///
+	/// - `who`: the AccountId whose borrow should be calculated.
+	pub fn get_user_total_borrow_usd(who: &T::AccountId) -> BalanceResult {
```
Comment (lut1k) [inline pallets/controller/src/lib.rs:604]: I think this function can be implemented in the `trait ControllerManager`

**Review by lut1k — DISMISSED** [review]

**Review by Apollo50 — DISMISSED** [review]

**Review by lut1k — DISMISSED** [review]

**Review by lut1k — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Review by kurotych — APPROVED** [review]

**Pull request merged**